addons/crm/models/crm_lead.py

```diff
@@ -967,106 +989,60 @@ def _create_lead_partner_data(self, partner_name, is_company=False, parent_id=Fa
+        partner on most leads; either through a newly created partner either
+        through a given partner_id.
 
-        :param string action: what has to be done regarding partners (create it, assign an existing one, or nothing)
-        :param int partner_id: partner to assign if any
-        :return: dict(lead_id: partner_id)
+        :param int force_partner_id: if set, update all leads to that customer;
+        :param create_missing: for leads without customer, create a new one
+          based on lead information;
         """
-        partner_ids = {}
         for lead in self:
-            if partner_id:
-                lead.partner_id = partner_id
-            if lead.partner_id:
-                partner_ids[lead.id] = lead.partner_id.id
-                continue
-            if action == 'create':
-                partner = lead._create_lead_partner()
-                partner_id = partner.id
-                partner.team_id = lead.team_id
-            partner_ids[lead.id] = partner_id
-        return partner_ids
-
-    def allocate_salesman(self, user_ids=None, team_id=False):
+            if force_partner_id:
+                lead.partner_id = force_partner_id
+            if not lead.partner_id and create_missing:
+                partner = lead._create_customer()
+                lead.partner_id = partner.id
+        return True
+
+    def handle_salesmen_assignment(self, user_ids=None, team_id=False):
         """ Assign salesmen and salesteam to a batch of leads.  If there are more
-            leads than salesmen, these salesmen will be assigned in round-robin.
-            E.g.: 4 salesmen (S1, S2, S3, S4) for 6 leads (L1, L2, ... L6).  They
-            will be assigned as followed: L1 - S1, L2 - S2, L3 - S3, L4 - S4,
-            L5 - S1, L6 - S2.
-
-            :param list ids: leads/opportunities ids to process
-            :param list user_ids: salesmen to assign
-            :param int team_id: salesteam to assign
-            :return bool
+        leads than salesmen, these salesmen will be assigned in round-robin. E.g.
+        4 salesmen (S1, S2, S3, S4) for 6 leads (L1, L2, ... L6) will assigned as
+        following: L1 - S1, L2 - S2, L3 - S3, L4 - S4, L5 - S1, L6 - S2.
+
+        :param list user_ids: salesmen to assign
+        :param int team_id: salesteam to assign
         """
-        index = 0
-        for lead in self:
-            value = {}
-            if team_id:
-                value['team_id'] = team_id
+        update_vals = {'team_id': team_id} if team_id else {}
+        for index, lead in enumerate(self):
             if user_ids:
-                value['user_id'] = user_ids[index]
-                # Cycle through user_ids
-                index = (index + 1) % len(user_ids)
-            if value:
-                lead.write(value)
+                update_vals['user_id'] = user_ids[index % len(user_ids)]
+            if update_vals:
+                lead.write(update_vals)
         return True
```

Comment:
Same here, I don't think it's useful to return something as the return value is never used

Reply:
Ok done